### PR TITLE
Add kube-vip field to TKR.spec.kubernetes

### PR DESCRIPTION
**Pull request by imikushin**
### What this PR does / why we need it

Add `kube-vip` field to spec.kubernetes in TanzuKubernetesRelease. A ClusterClass patch would inject the data from this field into kubeadm configuration, so kube-vip would run as a static pod.

### Describe testing done for PR

Updated unit-tests.

### Release note

```release-note
NONE
```

### PR Checklist

- [X] Squash the commits into one or a small number of logical commits
      
- [X] Use good commit [messages](https://github.com/vmware-tanzu/tanzu-framework/blob/main/CONTRIBUTING.md)
- [X] Ensure PR contains terms all contributors can understand and links all contributors can access

### CODE REVIEW

File: config/crd/bases/run.tanzu.vmware.com_tanzukubernetesreleases.yaml [hunk 1]
```diff
@@ -5,22 +5,9 @@ kind: CustomResourceDefinition
 metadata:
   annotations:
     controller-gen.kubebuilder.io/version: v0.7.0
-    cert-manager.io/inject-ca-from: tanzu-system/tkr-conversion-webhook-serving-cert
   creationTimestamp: null
   name: tanzukubernetesreleases.run.tanzu.vmware.com
 spec:
-  conversion:
```
Comment (vuil) [inline config/crd/bases/run.tanzu.vmware.com_tanzukubernetesreleases.yaml:12]: is this meant to be part of the change?

**Review by vuil — APPROVED** [review]
kgtm, just had a small question.

Also, nit: https://github.com/vmware-tanzu/tanzu-framework/pull/1874/files#diff-0cc91b87461b5d5629157a437ec53993f384290559af2d009d7c4a662115aa0cR236:
perhaps good to update the description on what the section is about?

**Review by tenczar — APPROVED** [review]
Looks good to me. I do have a question about the two changes from `storage: true` -> `storage: false`. Are these intended to be part of this change?

File: config/crd/bases/run.tanzu.vmware.com_tanzukubernetesreleases.yaml (hunk 1 above)
Comment (imikushin) [reply config/crd/bases/run.tanzu.vmware.com_tanzukubernetesreleases.yaml]: Fixed this in #1882

**Comment by imikushin** [comment]
> Looks good to me. I do have a question about the two changes from `storage: true` -> `storage: false`. Are these intended to be part of this change?

Changed this in #1882

**Pull request merged**